### PR TITLE
Explain ledger errors

### PR DESCRIPTION
**Pull request by lukaw3d**
Fixes https://github.com/oasisprotocol/oasis-wallet-web/issues/374
Improves https://github.com/oasisprotocol/oasis-wallet-web/issues/376

Not sure what to do about translations :shrug:

### CODE REVIEW

File: src/app/components/ErrorFormatter/index.tsx [hunk 1]
```diff
@@ -27,20 +28,34 @@ export function ErrorFormatter(props: Props) {
     [WalletErrors.InvalidNonce]: t('errors.invalidNonce'),
     [WalletErrors.DuplicateTransaction]: t('errors.duplicateTransaction'),
     [WalletErrors.NoOpenWallet]: t('errors.noOpenWallet'),
-    [WalletErrors.USBTransportError]: t('errors.usbTransportError', 'USB Transport error : {{message}}', {
-      message,
-    }),
+    [WalletErrors.USBTransportError]: t(
+      'errors.usbTransportError',
+      'USB Transport error : {{message}}. This usually means your browser is not supported - Chrome should work, but Firefox does not support WebUSB.',
```
Comment (matevz) [inline src/app/components/ErrorFormatter/index.tsx:33]: The space on both the left and right side of : is some French specialty.

```suggestion
      'USB Transport error: {{message}}. This usually means your browser does not support WebUSB. Try using a Google Chrome-based browser and check application permissions.',
```

File: src/app/components/ErrorFormatter/index.tsx [hunk 2]
```diff
@@ -27,20 +28,34 @@ export function ErrorFormatter(props: Props) {
...
+      {
+        message,
+      },
+    ),
     [WalletErrors.LedgerAppVersionNotSupported]: t(
       'errors.ledgerAppVersionNotSupported',
-      'Received : "Ledger App Version not supported", this can be a false positive, retry the operation, if the error persists, update the Oasis App on your Ledger.',
+      'Could not open Oasis App on Ledger, make sure it is unlocked and that the Oasis App is opened, and up to date.',
```
Comment (matevz) [inline src/app/components/ErrorFormatter/index.tsx:40]:
```suggestion
      'Could not open Oasis App on Ledger. Make sure Ledger is unlocked, the Oasis App is opened and up to date.',
```

File: src/app/components/ErrorFormatter/index.tsx [hunk 3]
```diff
@@ -27,20 +28,34 @@ export function ErrorFormatter(props: Props) {
...
       'Transaction rejected on Ledger',
     ),
-    [WalletErrors.LedgerNoDeviceSelected]: t(
-      'errors.ledgerNoDeviceSelected',
-      'No ledger selected. If you wish to open your ledger, close this message and retry.',
+    [WalletErrors.LedgerNoDeviceSelected]: (
+      <Trans
+        i18nKey="errors.ledgerNoDeviceSelected"
+        t={t}
+        defaults="No ledger device selected. If your device was not selectable, make sure it is connected and <0>fix any USB connection issues with Ledger</0>."
```
Comment (matevz) [inline src/app/components/ErrorFormatter/index.tsx:50]:
```suggestion
        defaults="No Ledger device selected. Make sure it is connected and <0>check common USB connection issues with Ledger</0>."
```

File: src/locales/en/translation.json [hunk 4]
```diff
@@ -131,10 +131,10 @@
     "invalidAddress": "Invalid address",
     "insufficientBalance": "Insufficient balance",
     "noOpenWallet": "",
-    "usbTransportError": "USB Transport error : {{message}}",
-    "ledgerAppVersionNotSupported": "Received : \"Ledger App Version not supported\", this can be a false positive, retry the operation, if the error persists, update the Oasis App on your Ledger.",
+    "usbTransportError": "USB Transport error : {{message}}. This usually means your browser is not supported - Chrome should work, but Firefox does not support WebUSB.",
```
Comment (matevz) [inline src/locales/en/translation.json:134]: Bump this as well, if you decide to accept my changes.

**Review by matevz — APPROVED** [review]

File: src/app/components/ErrorFormatter/index.tsx (hunk 2 above)
Comment (pro-wh) [reply src/app/components/ErrorFormatter/index.tsx:40]: but it really might be an unsupported version, right? should we keep something about the original message too?

**Review by pro-wh — APPROVED** [review]

File: src/app/components/ErrorFormatter/index.tsx (hunk 2 above)
Comment (lukaw3d) [reply src/app/components/ErrorFormatter/index.tsx:40]: I don't know how to install an old version to check what it throws in that case

Comment (lukaw3d) [reply src/app/components/ErrorFormatter/index.tsx]: How about:
```diff
- Could not open Oasis App on Ledger. Make..
+ Oasis App on Ledger is closed or outdated. Make..
```

File: src/app/components/ErrorFormatter/index.tsx (hunk 1 above)
Comment (lukaw3d) [reply src/app/components/ErrorFormatter/index.tsx]: I disagree. This is more accurate for like.. a linux user. But no one else will understand.

File: src/app/components/ErrorFormatter/index.tsx (hunk 2 above)
Comment (pro-wh) [reply src/app/components/ErrorFormatter/index.tsx]: ok

File: src/locales/en/translation.json [hunk 5]
```diff
@@ -131,11 +131,11 @@
...
-    "usbTransportError": "USB Transport error: {{message}}",
-    "ledgerAppVersionNotSupported": "Received: \"Ledger App Version not supported\", this can be a false positive, retry the operation, if the error persists, update the Oasis App on your Ledger.",
+    "usbTransportError": "USB Transport error: {{message}}. This usually means your browser does not support WebUSB. Try using a Chromium-based browser and check application permissions.",
+    "ledgerAppVersionNotSupported": "Could not open Oasis App on Ledger. Make sure Ledger is unlocked, the Oasis App is opened and up to date.",
     "ledgerTransactionRejected": "Transaction rejected on Ledger",
-    "ledgerNoDeviceSelected": "No ledger selected. If you wish to open your ledger, close this message and retry.",
-    "ledgerCannotOpenOasisApp": "Could not open Oasis App on Ledger, make sure it is unlocked and that the Oasis App is opened.",
+    "ledgerNoDeviceSelected": "No Ledger device selected. Make sure it is connected and <0>check common USB connection issues with Ledger</0>.",
+    "ledgerCannotOpenOasisApp": "Could not open Oasis App on Ledger. Make sure Ledger is unlocked and  the Oasis App is opened.",
```
Comment (pro-wh) [inline src/locales/en/translation.json:138]:
```suggestion
    "ledgerCannotOpenOasisApp": "Could not open Oasis App on Ledger. Make sure Ledger is unlocked and the Oasis App is opened.",
```

extra space

File: src/app/components/ErrorFormatter/index.tsx (hunk 1 above)
Comment (lukaw3d) [reply src/app/components/ErrorFormatter/index.tsx]: cc @tjanez

Comment (lukaw3d) [reply src/app/components/ErrorFormatter/index.tsx]: 'USB Transport error: {{message}}. This usually means your browser does not support WebUSB (e.g. Firefox). Try using Chrome and check application permissions.'

**Pull request merged**